packages/flutter_version_manager/lib/src/check_updates.dart

```diff
@@ -24,7 +23,8 @@ Future<void> checkForUpdates({
   VoidCallback? onUpdateEnd,
   bool backendLeading = true,
 }) async {
-  var expectedBackendVersion = await getKeyFromPubspecOrNull("backend-version");
+  var expectedBackendVersion =
+      (await service.getExpectedBackendVersion())?.toString();
```

Comment:
I think calling toString here should not be required. This should always be a nullable string. 

It's like an id on an object, generally it can be an int due to automated indexing on database. In reality, all id's should probably be strings, as they do not require actual calculations.

Reply:
I removed the .toString() and parsing it back to Version.
For now we are only dealing with Versions everywhere